pkg/model/manifest.go

```diff
@@ -1206,3 +1206,68 @@ func getBuildContextForComposeWithVolumeMounts(m *Manifest) (string, error) {
 
 	return context, nil
 }
+
+func (di DeployInfo) IsEmpty() bool {
+	if len(di.Commands) > 0 {
+		return false
+	}
+	if di.ComposeSection != nil && di.ComposeSection.Stack != nil {
+		return false
+	}
+	if di.Divert != nil {
+		return false
+	}
+	if di.Endpoints != nil {
+		return false
+	}
+	return true
+}
+
+func (di DestroyInfo) IsEmpty() bool {
+	if di.Image != "" {
+		return false
+	}
+	if len(di.Commands) > 0 {
+		return false
+	}
+	return true
+}
+
+func (mt ManifestTests) IsEmpty() bool {
+	return len(mt) == 0
+}
+
+func (m *Manifest) ValidateForCLIOnly() error {
+	invalidFields := []string{}
+	if !(m.Build == nil || m.Build.IsEmpty()) {
+		invalidFields = append(invalidFields, "build")
+	}
+	if !(m.Dependencies == nil || m.Dependencies.IsEmpty()) {
+		invalidFields = append(invalidFields, "dependencies")
+	}
+	if !(m.Deploy == nil || m.Deploy.IsEmpty()) {
+		invalidFields = append(invalidFields, "deploy")
+	}
+	if !(m.Destroy == nil || m.Destroy.IsEmpty()) {
+		invalidFields = append(invalidFields, "destroy")
+	}
+	if !(m.External == nil || m.External.IsEmpty()) {
+		invalidFields = append(invalidFields, "external")
+	}
+	if !(m.Test == nil || m.Test.IsEmpty()) {
+		invalidFields = append(invalidFields, "test")
+	}
+	if m.Icon != "" {
+		invalidFields = append(invalidFields, "icon")
+	}
+	if m.Name != "" {
+		invalidFields = append(invalidFields, "name")
+	}
+	if len(invalidFields) > 0 {
+		return oktetoErrors.UserError{
+			E:    fmt.Errorf("the following fields are not allowed in this context: %s", strings.Join(invalidFields, ", ")),
```

Comment:
I think the error message should not be saying "invalid" fields, but we should say something like:

"These features are only supported if you install Okteto in your cluster" or something along those lines

cc @codyjlandstrom can you help out here?

Reply:
I agree with your suggestion @andreafalzetti - then the user knows what the context is.

"The following features are only supported if you install Okteto in your cluster:"

Reply:
@codyjlandstrom FYI in various of places of the CLI we use the word "Context" instead of "Cluster". This is something product requested to change some time ago. To keep this consistency I suggest we keep "context" in this change and if we want to change the naming in the future, do it in all places with another PR.

@jLopezbarb WDYT?